arch/xtensa/src/common/xtensa.h

```diff
@@ -105,12 +105,9 @@
 
 /* In the XTENSA model, the state is copied from the stack to the TCB, but
  * only a referenced is passed to get the state from the TCB.
- *
- * REVISIT: It would not be too difficult to save only a pointer to the
- * state save area in the TCB and thus avoid the copy.
  */
 
-#define xtensa_savestate(regs)    xtensa_copystate(regs, (uint32_t*)CURRENT_REGS)
+#define xtensa_savestate(regs)    do { regs = (uint32_t *)CURRENT_REGS; } while (0)
 #define xtensa_restorestate(regs) do { CURRENT_REGS = regs; } while (0)
```

Comment:
```suggestion
#define xtensa_savestate(regs)    (regs = (uint32_t *)CURRENT_REGS)
#define xtensa_restorestate(regs) (CURRENT_REGS = regs)
```
nit: These defines are composed of a single statement, we could remove the `do {...} while(0)`.

Reply:
Done, thanks!